lib/connect.c

```diff
@@ -677,7 +677,7 @@ void Curl_updateconninfo(struct connectdata *conn, curl_socket_t sockfd)
 {
   if(conn->transport == TRNSPRT_TCP) {
 #if defined(HAVE_GETPEERNAME) || defined(HAVE_GETSOCKNAME)
-    if(!conn->bits.reuse && !conn->bits.tcp_fastopen) {
+    if(!conn->bits.reuse) {
```

Comment:
Possibly the reason this information isn't retrieved when fastopen is enabled is because it may not be available yet.

Reply:
Ok, when I remove this change from the PR without replacement, the PR will not solve the original issue.
The it would be necessary to postpone the detection of the address into the FTP part, when it is missing.
I backed off this approach, because it introduces some additional code.